service/history/execution/context.go

```diff
@@ -720,11 +719,12 @@ func (c *contextImpl) UpdateWorkflowExecutionWithNew(
 	if err != nil {
 		return err
 	}
-
 	var persistedBlobs events.PersistedBlobs
-	currentWorkflowSize := c.GetHistorySize()
+	currentWorkflowSize, oldWorkflowSize := c.GetHistorySize(), c.GetHistorySize()
+	currentWorkflowHistoryCount, oldWorkflowHistoryCount := c.mutableState.GetNextEventID()-1, c.mutableState.GetNextEventID()-1
```

Comment:
```suggestion
	currentWorkflowSize := c.GetHistorySize()
	oldWorkflowSize := currentWorkflowSize
	currentWorkflowHistoryCount := c.mutableState.GetNextEventID()-1
	oldWorkflowHistoryCount := currentWorkflowHistoryCount
```
since that's the intent, and it makes it impossible for it to mean anything different (e.g. say GetNextEventID() mutated something).